### PR TITLE
Handle server error in Collaborators.vue

### LINKED ISSUE #2365: Fix server error handling in frontend
Server errors are now coming in a slightly different format.

**Before:**
validation message were included in a key `validation_messages`
https://api-tools.getlaminas.org/documentation/api-primer/error-reporting

**Now:** 
validation messages are included in a key `violations`
https://api-platform.com/docs/core/validation/#validating-submitted-data

Needs to be fixed for every code that relies on existence of `validation_messages` key. Or check if general ServerError.vue component could be used.
At the moment this is utilized in
- ServerError.vue (fixed in #2366)
- Collaborators.vue (#2897)
- ApiWrapper.vue (fixed in #2482 )

### PR DESCRIPTION
**Pull request by BacLuc**
And also indicate if the value for inviteEmail is not a valid email.

closes #2365

### CODE REVIEW

File: frontend/src/views/camp/Collaborators.vue [hunk 1]
```diff
@@ -143,8 +146,8 @@ export default {
       return this.collaborators.filter((c) => c.status === 'inactive')
     },
     inviteEmailMessages() {
-      return this.messages.inviteEmail
-        ? Object.values({ ...this.messages.inviteEmail })
+      return this.messages
+        ? Object.values({ ...this.messages.map((value) => serverErrorToString(value)) })
```
Comment (usu) [inline frontend/src/views/camp/Collaborators.vue]: `serverErrorToString` expects as parameter the error object returned from `hal-json-vuex` and not the violations array. You can directly provide the `e` property from `handleError`. Check for example `DayResponsibles.vue` or `ActivityResponsibles.vue`.

In my opinion we could also move the 409 check into `serverErrorToString`, I guess this would improve error handling for other components, too.

**Review by usu — CHANGES_REQUESTED** [review]
Improvement idea: Clear server error once inviteEmail is modified by user.

File: frontend/src/views/camp/Collaborators.vue (hunk 1 above)
Comment (BacLuc) [reply frontend/src/views/camp/Collaborators.vue]: serverErrorToString only returns the error.response.data.detail property, which is a concatenated list of the errors.
I would prefer to have the validation errors as a structured object (e.g. a list), that currently vuetify (and maybe later we) can process the individual errors.

Using serverErrorToString on the e variable:
image

Using the validation messages in a structured format:
image

added :error-count="2" to show 2 errors in https://github.com/ecamp/ecamp3/pull/2897/commits/7bc9cf91a5292afe99bb98c5f66c93c6a8218406

Comment (BacLuc) [reply frontend/src/views/camp/Collaborators.vue]: As far as i know, we don't have checks for concurrent updates in the backend.
So the 409 error code will currently never occur.

Should i really move it?

Comment (usu) [reply frontend/src/views/camp/Collaborators.vue]: I'm fine with the list approach, but you're still misusing `serverErrorToString` which expects a complete error object but you throw an item of the violations array at it (which by accident contains a message property) 😀 

Tried & pushed an implementation:
https://github.com/ecamp/ecamp3/pull/2897/commits/b8330c3a984c9d3b278d07988597d8bd3c88fe3d

File: frontend/src/views/camp/Collaborators.vue [hunk 2]
```diff
@@ -56,7 +56,6 @@ Displays collaborators of a single camp.
                 <e-text-field
                   v-model="inviteEmail"
                   :name="$tc('views.camp.collaborators.email')"
-                  vee-rules="email"
```
Comment (BacLuc) [inline frontend/src/views/camp/Collaborators.vue]: Why was this removed?

File: frontend/src/views/camp/Collaborators.vue (hunk 1 above)
Comment (BacLuc) [reply frontend/src/views/camp/Collaborators.vue]: Thank you. :+1:

File: frontend/src/views/camp/Collaborators.vue (hunk 2 above)
Comment (usu) [reply frontend/src/views/camp/Collaborators.vue]: oops, shouldn't. Test code to trigger backend errors. Shall I put it back or will you?

Comment (BacLuc) [reply frontend/src/views/camp/Collaborators.vue]: i will put it back, no problem.

**Review by usu — APPROVED** [review]

**Review by carlobeltrame — APPROVED** [review]
